### PR TITLE
feat(checks-gcp): Include 4 new checks covering GCP CIS

### PR DESCRIPTION
**Pull request by jit-contrib**
### Context

Adding support for new GCP rules.

### Description

* Added 4 new GPC rules to gcp provider;
* Added unit tests to new created rules
* compute_service of gcp provider edited to include new data used in the new rules

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jfagoagas** [comment]
Hi @jit-contrib, thanks for this contribution. In the description you were saying "Added unit tests to new created rules" but I can't see any test pushed to the branch, is it okay? Maybe something was missed during the push.

Thanks!

File: prowler/providers/gcp/services/compute/compute_shielded_vm_enabled/compute_shielded_vm_enabled.metadata.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "ResourceType": "VMInstance",
+    "Description": "To defend against advanced threats and ensure that the boot loader and firmware on your VMs are signed and untampered, it is recommended that Compute instances are launched with Shielded VM enabled.",
+    "Risk": "",
+    "RelatedUrl": "",
+    "Remediation": {
+      "Code": {
+        "CLI": "",
+        "NativeIaC": "",
+        "Other": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/gcp/ComputeEngine/enable-shielded-vm.html",
+        "Terraform": ""
```
Comment (MrCloudSec) [inline prowler/providers/gcp/services/compute/compute_shielded_vm_enabled/compute_shielded_vm_enabled.metadata.json]:
```suggestion
    "Remediation": {
      "Code": {
        "CLI": "https://docs.bridgecrew.io/docs/bc_gcp_general_y#cli-command",
        "NativeIaC": "",
        "Other": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/gcp/ComputeEngine/enable-shielded-vm.html",
        "Terraform": "https://docs.bridgecrew.io/docs/bc_gcp_general_y#terraform"
```

File: prowler/providers/gcp/services/compute/compute_serial_ports_in_use/compute_serial_ports_in_use.metadata.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "ResourceType": "VMInstance",
+    "Description": "Interacting with a serial port is often referred to as the serial console, which is similar to using a terminal window, in that input and output is entirely in text mode and there is no graphical interface or mouse support. If you enable the interactive serial console on an instance, clients can attempt to connect to that instance from any IP address. Therefore interactive serial console support should be disabled.",
+    "Risk": "",
+    "RelatedUrl": "",
+    "Remediation": {
+      "Code": {
+        "CLI": "",
+        "NativeIaC": "",
+        "Other": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/gcp/ComputeEngine/disable-interactive-serial-console-support.html",
+        "Terraform": ""
```
Comment (MrCloudSec) [inline prowler/providers/gcp/services/compute/compute_serial_ports_in_use/compute_serial_ports_in_use.metadata.json]:
```suggestion
      "Code": {
        "CLI": "https://docs.bridgecrew.io/docs/bc_gcp_networking_11#cli-command",
        "NativeIaC": "",
        "Other": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/gcp/ComputeEngine/disable-interactive-serial-console-support.html",
        "Terraform": "https://docs.bridgecrew.io/docs/bc_gcp_networking_11#terraform"
```

File: prowler/providers/gcp/services/compute/compute_default_service_account_in_use_with_full_api_access/compute_default_service_account_in_use_with_full_api_access.metadata.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    "ResourceType": "VMInstance",
+    "Description": "To support principle of least privileges and prevent potential privilege escalation it is recommended that instances are not assigned to default service account `Compute Engine default service account` with Scope `Allow full access to all Cloud APIs`.",
+    "Risk": "",
+    "RelatedUrl": "",
+    "Remediation": {
+      "Code": {
+        "CLI": "",
+        "NativeIaC": "",
+        "Other": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/gcp/ComputeEngine/default-service-accounts-with-full-access-in-use.html",
+        "Terraform": ""
```
Comment (MrCloudSec) [inline prowler/providers/gcp/services/compute/compute_default_service_account_in_use_with_full_api_access/compute_default_service_account_in_use_with_full_api_access.metadata.json]:
```suggestion
        "CLI": "https://docs.bridgecrew.io/docs/bc_gcp_iam_2#cli-command",
        "NativeIaC": "",
        "Other": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/gcp/ComputeEngine/default-service-accounts-with-full-access-in-use.html",
        "Terraform": "https://docs.bridgecrew.io/docs/bc_gcp_iam_2#terraform"
```

File: prowler/providers/gcp/services/compute/compute_default_service_account_in_use/compute_default_service_account_in_use.metadata.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    "ResourceType": "VMInstance",
+    "Description": "It is recommended to configure your instance to not use the default Compute Engine service account because it has the Editor role on the project.",
+    "Risk": "",
+    "RelatedUrl": "",
+    "Remediation": {
+      "Code": {
+        "CLI": "",
+        "NativeIaC": "",
+        "Other": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/gcp/ComputeEngine/default-service-accounts-in-use.html",
+        "Terraform": ""
```
Comment (MrCloudSec) [inline prowler/providers/gcp/services/compute/compute_default_service_account_in_use/compute_default_service_account_in_use.metadata.json]:
```suggestion
      "Code": {
        "CLI": "https://docs.bridgecrew.io/docs/bc_gcp_iam_1#cli-command",
        "NativeIaC": "",
        "Other": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/gcp/ComputeEngine/default-service-accounts-in-use.html",
        "Terraform": "https://docs.bridgecrew.io/docs/bc_gcp_iam_1#terraform"
```

**Review by MrCloudSec — APPROVED** [review]
Awesome job!

**Pull request merged**